Cryptography.java

```diff
@@ -0,0 +1,174 @@
+import javax.crypto.Cipher;
+import javax.crypto.Mac;
+import javax.crypto.spec.IvParameterSpec;
+import javax.crypto.spec.SecretKeySpec;
+import java.security.spec.*;
+import java.security.*;
+import java.io.*;
+import java.util.*;
+import java.nio.file.*;
+
+public class Cryptography {
+	static private int ivSize = 16;
+	static private int keySize = 16;
+
+	private static PublicKey readPublicKey(Path path) throws IOException, NoSuchAlgorithmException, InvalidKeySpecException {
+	    X509EncodedKeySpec publicSpec = new X509EncodedKeySpec(Files.readAllBytes(path));
+	    KeyFactory keyFactory = KeyFactory.getInstance("RSA");
+	    return keyFactory.generatePublic(publicSpec);       
+	}
+
+	private static PrivateKey readPrivateKey(Path path) throws IOException, NoSuchAlgorithmException, InvalidKeySpecException {
+	    PKCS8EncodedKeySpec keySpec = new PKCS8EncodedKeySpec(Files.readAllBytes(path));
+	    KeyFactory keyFactory = KeyFactory.getInstance("RSA");
+	    return keyFactory.generatePrivate(keySpec);     
+	}
+
+	public static Boolean compareMAC(byte[] m1, byte[] m2) throws IOException {
+		if (!Arrays.equals(m1, m2))
+			return false;
+		return true;
+	}
+
+	public static byte[] generateMAC(String message, String key) throws IOException {
+		try {
+			//Cast key to a byte array and generate a SecretKeySpec needed for mac.init
+			SecretKeySpec keySpec = new SecretKeySpec(key.getBytes(), "AES");
+
+			// Generate MAC
+			Mac mac = Mac.getInstance("HmacSHA256");
+			mac.init(keySpec);
+			byte[] result = mac.doFinal(message.getBytes());
+			return result;
+		} catch (Exception e) {
+			System.out.println("Error in AES.generateMAC: " + e);
+			return null;
+		}
+	}
+
+	public static byte[] digestMessage(byte[] message) {
+		try{
+			MessageDigest digest = MessageDigest.getInstance("SHA-256");
+			digest.update(message);
+			byte[] messageDigest = digest.digest();
+			return messageDigest;
+		}
+		catch (Exception e) {
+			System.out.println("Error in AES.digestMessage: " + e);
+			return null;
+		}
+	}
+
+	public static Boolean compareDigests(byte[] d1, byte[] d2) {
+		if (!Arrays.equals(d1, d2)) return false;
+		return true;
+	}
+
+	public static byte[] createSignature(byte[] digest, Path privateKeyPath) {
+		try{
+			PrivateKey privateKey = readPrivateKey(privateKeyPath);
+			Cipher cipher = Cipher.getInstance("RSA");
+			cipher.init(Cipher.ENCRYPT_MODE, privateKey);
```

Comment:
Hey, im not 100% sure on this, i will double check but I think instead of signing you are encrypting and decrypting